events-processor/config/kafka/consumer.go

```diff
@@ -122,23 +126,28 @@ func (cg *ConsumerGroup) lost(_ context.Context, _ *kgo.Client, lost map[string]
 
 func (cg *ConsumerGroup) poll() {
 	for {
-		fetches := cg.client.PollRecords(context.Background(), 10000)
-		if fetches.IsClientClosed() {
-			cg.logger.Info("client closed")
-			return
-		}
+		// NOTE: Gracefull shutdown handling will be plugged in here
+		cg.pollRecords()
+	}
+}
+
+func (cg *ConsumerGroup) pollRecords() {
+	fetches := cg.client.PollRecords(context.Background(), 10000)
+	if fetches.IsClientClosed() {
+		cg.logger.Info("client closed")
+		return
```

Comment:
@vincent-pochet this is not the same behaviour. Before we returned from `poll` when client got closed. Now we will loop indefinitely.